src/NexusMods.App.UI/Pages/Downloads/DownloadsPage.cs

```diff
@@ -0,0 +1,39 @@
+using JetBrains.Annotations;
+using Microsoft.Extensions.DependencyInjection;
+using NexusMods.Abstractions.Downloads;
+using NexusMods.Abstractions.Games;
+using NexusMods.Abstractions.NexusWebApi.Types.V2;
+using NexusMods.Abstractions.Serialization.Attributes;
+using NexusMods.App.UI.Windows;
+using NexusMods.App.UI.WorkspaceSystem;
+
+namespace NexusMods.App.UI.Pages.Downloads;
+
+[JsonName("NexusMods.App.UI.Pages.Downloads.DownloadsPageContext")]
+public record DownloadsPageContext : IPageFactoryContext;
+
+[JsonName("NexusMods.App.UI.Pages.Downloads.AllDownloadsPageContext")]
+public record AllDownloadsPageContext : DownloadsPageContext;
+
+[JsonName("NexusMods.App.UI.Pages.Downloads.CompletedDownloadsPageContext")]
+public record CompletedDownloadsPageContext : DownloadsPageContext;
+
+[JsonName("NexusMods.App.UI.Pages.Downloads.GameSpecificDownloadsPageContext")]
+public record GameSpecificDownloadsPageContext(GameId GameId) : DownloadsPageContext;
```

Comment:
Do you plan to make 4 separate pages or are these just meant to be aliases for the same page using different filters?

Reply:
This is further modified in further PRs; so comment on it on the last one.
But in short, it's something forced by our workspace system.

I needed to use different classes for restoring the page after closing and re-opening.

Reply:
Just different context, no?

Reply:
Yeah; the actual `.axaml` page is reused.